### ISSUE TITLE
Abstracted APIs for interoperability 

### ISSUE THREAD

**Issue by sfuji822** [hyperledger-cacti/cacti#418]
**Is your feature request related to a problem? Please describe.**
Considering the goal of our project, absorbing the difference of DLTs are high priority issue to be solved. 

**Describe the solution you'd like**
Define APIs for abstracted operating, in querying and monitoring the Ledger 

**Additional context**
At least three types of Ledgers are supported by CACTUS Ledger plugins for ensuring the feasibility of proposed APIs.

**Comment by RafaelAPB** [comment]
@sfuji822 This is an important issue, definitely agree with this. 
@petermetz, @hartm, this is actually (part) of the research task we have ahead of us when we will discuss the privacy framework for cross-chain transactions

**Comment by petermetz** [comment]
> @petermetz, @hartm, this is actually (part) of the research task we have ahead of us when we will discuss the privacy framework for cross-chain transactions

@RafaelAPB  Agreed, 100%. Hopefully we'll be at a stage soon where we at least have the basic mechanics for the initial set of supported ledgers nailed down (Corda is still heavily work in progress and the contract deployment needs to be worked out for Fabric).
Once these lower level primitives (e.g. the connectors) are ready and backed up by test automation, we'll be in a much stronger position to start evaluating how to make the API surfaces converge (optimally to reach complete equivalence, but the more I work on it the more that seems like a pipe dream TBH :-) ).
